Remove support for gfx940 and gfx941

JonathanLichtnerAMD opened the pull request:
This PR removes gfx940 and gfx941 from MIOpen.  Note that there is one "TODO" comment with gfx940 still in the code, but that will be left in and handled as part of a separate cleanup.

JonathanLichtnerAMD committed:
Remove support for gfx940 and gfx941

Note that there is still one comment with gfx940 left in the code, since
that cleanup will be handled by a different commit.

JonathanLichtnerAMD committed:
Remove kernel argument FORCE_CACHE_BYPASS_ON_STORE

This is always false now, so is no longer needed.

JonathanLichtnerAMD committed:
Cleanup places where "gfx94" is used in the code

There is only gfx942 now so this string can be replaced with
"gfx942" and the code updated.

Note that this commit has a side effect of fixing a bug in
test/gtest/conv_hip_igemm_xdlops.cpp, which was causing tests
to be skipped on gfx942 since it was comparing dev name to "gfx94".

JonathanLichtnerAMD committed:
Merge branch 'develop' into jlichtne/remove_gfx940-1

JonathanLichtnerAMD committed:
Merge branch 'develop' into jlichtne/remove_gfx940-1

JonathanLichtnerAMD committed:
Fix formatting after latest merge

averinevg commented on src/composable_kernel/composable_kernel/include/utility/config.hpp:
```diff
@@ -14,11 +14,10 @@
-      defined(CK_AMD_GPU_GFX950) || defined(CK_AMD_GPU_GFX1031) || defined(CK_AMD_GPU_GFX1100) ||  \
-      defined(CK_AMD_GPU_GFX1101) || defined(CK_AMD_GPU_GFX1102) || defined(CK_AMD_GPU_GFX1200) || \
-      defined(CK_AMD_GPU_GFX1201))
+      defined(CK_AMD_GPU_GFX908) || defined(CK_AMD_GPU_GFX90A) || defined(CK_AMD_GPU_GFX942) ||    \
+      defined(CK_AMD_GPU_GFX1030) || defined(CK_AMD_GPU_GFX950) || defined(CK_AMD_GPU_GFX1031) ||  \
+      defined(CK_AMD_GPU_GFX1100) || defined(CK_AMD_GPU_GFX1101) || defined(CK_AMD_GPU_GFX1102) || \
+      defined(CK_AMD_GPU_GFX1200) || defined(CK_AMD_GPU_GFX1201))
```
In my opinion, the changes in this file can be considered optional. Since the list was changed it could be sorted as well.

JonathanLichtnerAMD replied:
Sorted

averinevg commented on src/gemm_v2.cpp:
```diff
@@ -581,7 +581,7 @@ static void call_miopen_hipblasLt_gemm(const miopen::Handle& handle,
     break;
     case miopenFloat8_fnuz: {
-        const auto is_gfx94x = miopen::StartsWith(handle.GetDeviceName(), "gfx94");
+        const auto is_gfx94x = handle.GetDeviceName() == "gfx942";
```
[optional] You can also fix the comment in the exception below.

JonathanLichtnerAMD replied:
Done

averinevg commented on src/gemm_v2.cpp:
```diff
@@ -581,7 +581,7 @@ static void call_miopen_hipblasLt_gemm(const miopen::Handle& handle,
     break;
     case miopenFloat8_fnuz: {
-        const auto is_gfx94x = miopen::StartsWith(handle.GetDeviceName(), "gfx94");
+        const auto is_gfx94x = handle.GetDeviceName() == "gfx942";
```
```suggestion
        const auto is_gfx942 = handle.GetDeviceName() == "gfx942";
```

JonathanLichtnerAMD replied:
Done

averinevg commented on src/hip/batched_transpose_sol.cpp:
```diff
@@ -132,7 +132,7 @@ static inline const std::vector<BatchedTransposeParam>& GetKernelList(const Exec
         const auto device_name = ctx.GetStream().GetDeviceName();
-        if(StartsWith(device_name, "gfx94"))
+        if(device_name == "gfx942")
             return half_kernel_list_gfx94x;
```
```suggestion
            return half_kernel_list_gfx942;
```

JonathanLichtnerAMD replied:
Done

averinevg commented on src/include/miopen/solver/ck_utility_common.hpp:
```diff
@@ -54,9 +54,7 @@ static inline bool is_ck_supported_hardware(const Handle& handle)
            StartsWith(handle.GetDeviceName(), "gfx906") ||
            StartsWith(handle.GetDeviceName(), "gfx908") ||
            StartsWith(handle.GetDeviceName(), "gfx90a") ||
-           StartsWith(handle.GetDeviceName(), "gfx940") ||
            StartsWith(handle.GetDeviceName(), "gfx950") ||
-           StartsWith(handle.GetDeviceName(), "gfx941") ||
```
This needs to be sorted

JonathanLichtnerAMD replied:
I've sorted these, but they were not introduced by my changes

averinevg commented on src/include/miopen/solver/ck_utility_common.hpp:
```diff
@@ -108,12 +105,8 @@ static inline auto get_ck_common_compiler_flag(const Handle& handle)
-    else if(StartsWith(device_name, "gfx941"))
-        compiler_flag << " -DCK_AMD_GPU_GFX941";
     else if(StartsWith(device_name, "gfx942"))
         compiler_flag << " -DCK_AMD_GPU_GFX942";
```
This needs to be sorted

averinevg commented on src/include/miopen/solver/implicitgemm_util.hpp:
```diff
@@ -213,7 +213,7 @@ static inline bool IsXdlopsSupport(const ExecutionContext& ctx)
     const bool is_xdlops_supported = StartsWith(ctx.GetStream().GetDeviceName(), "gfx908") ||
                                      StartsWith(ctx.GetStream().GetDeviceName(), "gfx90a") ||
-                                     StartsWith(ctx.GetStream().GetDeviceName(), "gfx94");
+                                     StartsWith(ctx.GetStream().GetDeviceName(), "gfx942");
```
```suggestion
                                     ctx.GetStream().GetDeviceName() == "gfx942";
```

For other gpus here this can be changed as well

averinevg commented on src/include/miopen/solver/implicitgemm_util.hpp:
```diff
@@ -579,7 +579,7 @@ static inline bool IsComposableKernelSupportedHardware(const ExecutionContext& c
            StartsWith(c.GetStream().GetDeviceName(), "gfx908") ||
            StartsWith(c.GetStream().GetDeviceName(), "gfx90a") ||
-           StartsWith(c.GetStream().GetDeviceName(), "gfx94") ||
+           StartsWith(c.GetStream().GetDeviceName(), "gfx942") ||
```
`StartsWith()` can be removed, we can simply compare the strings for all gpus here (except gfx103X)

JonathanLichtnerAMD replied:
Done

averinevg commented on src/solver/conv/conv_winoRxS.cpp:
```diff
@@ -680,7 +680,7 @@ static bool IsApplicableBase(const ExecutionContext& ctx, const ProblemDescripti
     if(problem.IsFp16() &&
        !(StartsWith(name, "gfx906") || StartsWith(name, "gfx908") || StartsWith(name, "gfx90a") ||
-         StartsWith(name, "gfx94") || StartsWith(name, "gfx95") || StartsWith(name, "gfx1011") ||
+         StartsWith(name, "gfx942") || StartsWith(name, "gfx95") || StartsWith(name, "gfx1011") ||
```
We can compare the strings for all gpus here (except gfx103X and gfx11XX)

averinevg commented on test/gtest/conv_hip_igemm_xdlops.cpp:
```diff
@@ -89,7 +89,7 @@ void Run2dDriver(miopenDataType_t prec)
 {
     std::string devName = handle.GetDeviceName();
-    if(devName == "gfx908" || devName == "gfx90a" || devName == "gfx94")
+    if(devName == "gfx908" || devName == "gfx90a" || devName == "gfx942")
```
⚠️This test has never worked for gfx94X the way the author intended. It's worth paying attention to it.

JonathanLichtnerAMD replied:
Did you want me to do something here other than just make sure all the tests pass?

averinevg replied:
Just make sure all the tests pass on gfx942

averinevg commented on test/gtest/conv_trans.cpp:
```diff
@@ -89,7 +89,7 @@ bool IsTestSupportedForDevice(const miopen::Handle& handle)
     std::string devName = handle.GetDeviceName();
     if(devName == "gfx900" || devName == "gfx906" || devName == "gfx908" || devName == "gfx90a" ||
-       miopen::StartsWith(devName, "gfx94") || miopen::StartsWith(devName, "gfx103") ||
+       miopen::StartsWith(devName, "gfx942") || miopen::StartsWith(devName, "gfx103") ||
```
```suggestion
       devName == "gfx942" || miopen::StartsWith(devName, "gfx103") ||
```

averinevg commented on test/gtest/layernorm.cpp:
```diff
@@ -48,7 +48,7 @@ TEST_P(GPU_LayerNormTest_FP32, LayerNormTestFw)
     if((miopen::StartsWith(handle.GetDeviceName(), "gfx908") ||
         miopen::StartsWith(handle.GetDeviceName(), "gfx90a") ||
-        miopen::StartsWith(handle.GetDeviceName(), "gfx94")))
+        miopen::StartsWith(handle.GetDeviceName(), "gfx942")))
```
```suggestion
        handle.GetDeviceName() == "gfx942"))
```

averinevg reviewed (commented):
All my comments, except for one related to the test, are optional.

JonathanLichtnerAMD committed:
Merge branch 'develop' of github.com:ROCm/MIOpen into jlichtne/remove_gfx940-1

JonathanLichtnerAMD committed:
Rename is_gfx94x to is_gfx942 and related code

JonathanLichtnerAMD committed:
Rename half_kernel_list_gfx94x to half_kernel_list_gfx942

JonathanLichtnerAMD committed:
Sort a few places in the code where gfx950 is out of order

JonathanLichtnerAMD committed:
Remove some unnecessary use of StartsWith

JonathanLichtnerAMD committed:
Reformat code

averinevg reviewed (approved):
LGTM

BrianHarrisonAMD commented on src/kernels/conv1x1u_stride2.s:
```diff
@@ -936,10 +936,10 @@ loop_end:
+                    mbuffer_store_dwordx acc_cnt_\@-1, acc_ptr_\@+1, \v_offset, \s_desc, \s_offset, i_off\@ + 4
                 .else
-                    m_buffer_store_dwordx acc_cnt_\@, acc_ptr_\@, \v_offset, \s_desc, \s_offset, i_off\@
+                    mbuffer_store_dwordx acc_cnt_\@, acc_ptr_\@, \v_offset, \s_desc, \s_offset, i_off\@
```
Why did these change?

(m_buffer_store_dwordx -> mbuffer_store_dwordx)

JonathanLichtnerAMD replied:
This was inadvertently changed when I used search and replace to change _buffer_store_dwordx to buffer_store_dwordx.  I've reverted this part of the change.  Thanks for catching this!

JonathanLichtnerAMD replied:
Note that I ended up reverting these changes by reverting the commit "Remove kernel argument FORCE_CACHE_BYPASS_ON_STORE".  The change in that commit had removed a kernel build parameter that was always false;  however, this caused the following smoke tests to fail
```
Smoke/CPU_DBSync_NONE.StaticFDBSync/1
Smoke/CPU_DBSync_NONE.StaticFDBSync/2
```
since they were unable to find the kernel anymore in the database, as given by the following:

`KDB entry not found for  fdb-key:512-59-59-3x3-512-117-117-64-1x1-2x2-1x1-0-NCHW-FP32-B_g32, Solver: ConvBinWinogradRxSf2x3 perf config:10 filename: "Conv_Winograd_v30_3_1_fp32_f2x3_dilation2.s.o" compile args: -Wa,-defsym,ROCM_METADATA_VERSION=5 -mcumode -mwavefrontsize64 -mcpu=gfx90a
`

Since this change was a minor cleanup and not critical to the PR, I decided to remove it.  We can potentially clean this up in the future.

averinevg replied:
@JonathanLichtnerAMD All `FORCE_CACHE_BYPASS_ON_STORE` related changes were good, you just need to keep [this](https://github.com/ROCm/MIOpen/blob/c085744296a49f13c1b8b2317c989806763091aa/src/solver/conv/conv_winoRxS.cpp#L865) and [this](https://github.com/ROCm/MIOpen/blob/c085744296a49f13c1b8b2317c989806763091aa/src/solver/conv_winoRxS_fused.cpp#L228) lines.

JonathanLichtnerAMD replied:
Yes, I had considered that but went with removing the entire commit.  Anyway, I've reverted the revert and have added FORCE_CACHE_BYPASS_ON_STORE back with a value of false.

BrianHarrisonAMD reviewed (approved):
Looks good to me.

JonathanLichtnerAMD committed:
Rename mbuffer_store_dwordx back to m_buffer_store_dwordx

This was inadvertently changed by a search and replace of
_mbuffer_store_dwordx to buffer_store_dwordx.

JonathanLichtnerAMD committed:
Merge branch 'develop' of github.com:ROCm/MIOpen into jlichtne/remove_gfx940-1

JonathanLichtnerAMD committed:
Revert one last usage of mbuffer_store_short (to m_buffer_store_short)

JonathanLichtnerAMD committed:
Re-add FORCE_CACHE_BYPASS_ON_STORE back as a kernel build arg

This is needed or dbsync unit tests will fail.

JonathanLichtnerAMD committed:
Revert "Revert one last usage of mbuffer_store_short (to m_buffer_store_short)"

This reverts commit b4efd73b24885ee99f1cad6bf1a2bc6e0ed2eb6a.

JonathanLichtnerAMD committed:
Revert "Rename mbuffer_store_dwordx back to m_buffer_store_dwordx"

This reverts commit 14963a0f8b01f46d1ce905fdb044669ac5d13d35.

JonathanLichtnerAMD committed:
Revert "Remove kernel argument FORCE_CACHE_BYPASS_ON_STORE"

This reverts commit 30e81a0ab0ff21800837be07ee50d8cd0c0ae331.

JonathanLichtnerAMD committed:
Merge branch 'develop' of github.com:ROCm/MIOpen into jlichtne/remove_gfx940-1

JonathanLichtnerAMD committed:
Revert "Revert "Remove kernel argument FORCE_CACHE_BYPASS_ON_STORE""

This reverts commit f05f07dea1b525cb569a2f46459df216ef4a43a5.

JonathanLichtnerAMD committed:
Revert "Revert "Rename mbuffer_store_dwordx back to m_buffer_store_dwordx""

This reverts commit 98d5f320dbf8e44c5c3ac4b20b07c3feff0884a5.

JonathanLichtnerAMD committed:
Revert "Revert "Revert one last usage of mbuffer_store_short (to m_buffer_store_short)""

This reverts commit ff58ca319ed4f1032d56fbd046f2038510ba4b06.

JonathanLichtnerAMD committed:
Merge branch 'develop' into jlichtne/remove_gfx940-1
